Keanweng/inbox

keanweng97 opened the pull request:
Resolves #48 
Only has send Post functionality as there are no Like and Follow models.

Adding inbox (should not need to be manually created as Inbox is created whenever an Author is created):
![image](https://user-images.githubusercontent.com/35179885/108610246-75e00580-7391-11eb-8442-7cce74ff133b.png)

Inbox (will add Like and Follow objects later on):
![image](https://user-images.githubusercontent.com/35179885/108610263-a758d100-7391-11eb-89dc-a6b66d420ea6.png)

HailanXyouknow commented on server/inbox/migrations/0004_auto_20210219_2025.py:
```diff
@@ -0,0 +1,21 @@
+            name='author',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL),
+        ),
+    ]
```
Just a nit: I would delete all four migration files and regenerate the file so that only one file is pushed ;)

keanweng97 replied:
Didn't know you could do that, thanks!

HailanXyouknow commented on server/inbox/serializers.py:
```diff
@@ -0,0 +1,19 @@
+    author = serializers.SerializerMethodField()
+
+    def get_items(self, obj):
+        data = PostSerializer(obj.posts.all(), many=True).data
```
Nice nice :D

HailanXyouknow reviewed (approved):
Great work! 👍

keanweng97 commented:
Updated Inbox model, proper implementation of 'items' using ArrayField containing JSON of Post objects. Adding 'like' and 'follow' will be easier with this implementation.
![image](https://user-images.githubusercontent.com/35179885/109113971-436c3a80-76fa-11eb-983b-2ffe2c350965.png)
Added pagination too.

IanSeng commented on server/inbox/serializers.py:
```diff
@@ -0,0 +1,33 @@
+            data.append(literal_eval(item))
+        return data
+
+    class Meta:
```
do you mind moving putting Meta first, moving it to line 12

keanweng97 replied:
DRF has a similar format here: https://www.django-rest-framework.org/api-guide/serializers/#how-hyperlinked-views-are-determined

IanSeng reviewed (dismissed):
LGTM.

IanSeng reviewed (approved):
LGTM ✨
